pallets/process-validation/src/restrictions.rs

```diff
@@ -33,10 +50,23 @@ where
     T: Parameter + Default + Ord,
     V: Parameter + Default,
 {
-    match *restriction {
-        Restriction::None => true, // TODO implement some actual restrictions
-        Restriction::FixedNumberOfInputs { num_inputs } => return inputs.len() == num_inputs as usize,
-        Restriction::FixedNumberOfOutputs { num_outputs } => return outputs.len() == num_outputs as usize,
+    match &*restriction {
+        Restriction::<T, V>::None => true,
+        Restriction::FixedNumberOfInputs { num_inputs } => return inputs.len() == *num_inputs as usize,
+        Restriction::FixedNumberOfOutputs { num_outputs } => return outputs.len() == *num_outputs as usize,
+        Restriction::FixedMetadataValue {
+            input_index,
+            metadata_key,
+            metadata_value,
+        } => {
+            let selected_input = &inputs[*input_index as usize];
```

Comment:
I tried to use a match here but it opened a can of worms, 'a bool was expected instead of a unit' etc

Reply:
that's because you weren't returning but I think this can be even simpler. Could you not simply have:

```rust
let selected_input = &inputs[input_index as usize];
let meta = selected_input.metadata.get(&metadata_key);
meta == Some(&metadata_value)
```